### DISCUSSION TITLE
Features inheritance

### DISCUSSION THREAD

**Discussion by Trigve** [oleg-shilo/wixsharp#1618] [Ideas]
I would like to ask, if it would be possible to inherit the feature from the parent entities, when declaring the dirs/files. For instance, simplified example (the correct dirs/files are computed more complicated in real life):
```
...
new Dir(feat_postgres,
	"Sql",
	new Dir(new Id("postgres"),
		feat_postgres,
		"12",
		new Files(feat_postgres,
			@"src\sql\*.*"
		)
	)
),
```
As you can see, I need to add the same feature (`feat_postgres`) multiple times. What I would like to specifiy the feature in `Dir` entity only, and all children will ihnerit the feature.

**Comment by oleg-shilo** [comment]
It's problematic. Because DIR may have its own features that are not necessarily the same as the features of his children.

But you can definitely do this:
```C#
project.AllDirs
    .FirstOrDefault(x => x.Id != "your dir ID")
    .Files
    .ForEach(x=>x.Feature = yourFeature);

project.BuildMsi();
```
